File: unittest/t_no_time.py
#!/usr/bin/python
import sys
# import nwb
import test_utils as ut

from nwb import nwb_file
from nwb import nwb_utils as utils

# creates time series without 'timestamps' or 'starting_time' fields
# TESTS TimeSeries.ignore_time()
# TESTS timeseries placement in acquisition, stimulus, templates

def test_notime_series():
    if __file__.startswith("./"):
        fname = "s" + __file__[3:-3] + ".nwb"
    else:
        fname = "s" + __file__[1:-3] + ".nwb"
    name = "notime"
    # create_notime_series(fname, name, "acquisition")
    create_notime_series(fname, name, "/acquisition/timeseries")
    ut.verify_timeseries(fname, name, "acquisition/timeseries", "TimeSeries")
    ut.verify_absent(fname, "acquisition/timeseries/"+name, "timestamps")
    ut.verify_absent(fname, "acquisition/timeseries/"+name, "starting_time")

    # create_notime_series(fname, name, "stimulus")
    create_notime_series(fname, name, "/stimulus/presentation")
    ut.verify_timeseries(fname, name, "stimulus/presentation", "TimeSeries")
    # create_notime_series(fname, name, "template")
    create_notime_series(fname, name, "/stimulus/templates")
    ut.verify_timeseries(fname, name, "stimulus/templates", "TimeSeries")

def create_notime_series(fname, name, target):
    settings = {}
    # settings["filename"] = fname
    settings["file_name"] = fname
    # settings["identifier"] = nwb.create_identifier("notime example")
    settings["identifier"] = utils.create_identifier("notime example")
    # settings["overwrite"] = True
    settings["mode"] = "w"
    settings["start_time"] = "Sat Jul 04 2015 3:14:16"
    settings["description"] = "Test no time"
    # neurodata = nwb.NWB(**settings)
    f = nwb_file.open(**settings)
    
    
    #
#     notime = neurodata.create_timeseries("TimeSeries", name, target)
#     notime.ignore_time()
#     notime.set_data([0], unit="n/a", conversion=1, resolution=1)

    notime = f.make_group("<TimeSeries>", name, path=target)
    # following used for testing more missing_fields
    # notime = f.make_group("<VoltageClampSeries>", name, path=target)
    notime.set_dataset("data", [0], attrs={"unit":"n/a",
        "conversion":1, "resolution":1})

    #
    # notime.finalize()
    # neurodata.close()
    f.close()

test_notime_series()
print("%s PASSED" % __file__)

File: matlab_bridge/matlab_unittest/t_no_time.m
function t_no_time(verbosity)

% creates time series without 'timestamps' or 'starting_time' fields
% TESTS TimeSeries.ignore_time()
% TESTS timeseries placement in acquisition, stimulus, templates

script_base_name = mfilename();
script_name = [script_base_name '.m'];
fname = [regexprep(script_base_name, '^t_', 's_') '.nwb'];
if nargin < 1
    % default, display all information. Other options are: 'none', 'summary'
    verbosity = 'all';
end

function test_notime_series()
    name = 'notime';
    % create_notime_series(fname, name, 'acquisition')
    create_notime_series(fname, name, '/acquisition/timeseries')
    test_utils.verify_timeseries(fname, name, 'acquisition/timeseries', 'TimeSeries')
    test_utils.verify_absent(fname, ['acquisition/timeseries/',name], 'timestamps')
    test_utils.verify_absent(fname, ['acquisition/timeseries/',name], 'starting_time')

    % create_notime_series(fname, name, 'stimulus')
    create_notime_series(fname, name, '/stimulus/presentation')
    test_utils.verify_timeseries(fname, name, 'stimulus/presentation', 'TimeSeries')
    % create_notime_series(fname, name, 'template')
    create_notime_series(fname, name, '/stimulus/templates')
    test_utils.verify_timeseries(fname, name, 'stimulus/templates', 'TimeSeries')
end

function create_notime_series(fname, name, target)
    settings = {'file_name', fname, 'mode', 'w', 'verbosity', verbosity, ...
        'start_time', 'Sat Jul 04 2015 3:14:16' ...
        'identifier', nwb_utils.create_identifier('notime example'), ...
        'description','Test no time'};
    f = nwb_file(settings{:});

    notime = f.make_group('<TimeSeries>', name, 'path', target);
    % following used for testing more missing_fields
    % notime = f.make_group('<VoltageClampSeries>', name, path=target)
    % not sure why cell array used below.  Maybe required if only one
    % element
    notime.set_dataset('data', {0.0}, 'attrs', {'unit', 'n/a', ...
        'conversion', 1.0, 'resolution', 1.0});
    f.close()
end

test_notime_series()
fprintf('%s PASSED\n', script_name);
end

